---
title: "Week 5 Progress"
subtitle: "Quantifying DGE in *M. galloprovincialis* utilizing existing RNASeq data"
author: " C. Mantegna"
date: "`r format(Sys.time(), '%d %B, %Y')`"
format: 
  revealjs: 
    theme: serif
    navigation-mode: linear
    slide-number: true
    show-slide-number: print
editor: visual
---

## Project Overview {.smaller}

Utilizing differential gene expression to support conclusions about what common or unique physiological responses are displayed by *M. galloprovincialis* when exposed to the anthropogenic stressors listed below.  



| Accession ID | Tissue Type | Exposure                        |
|:------------:|:-----------:|:-------------------------------:|
| SRR19782039  | Gill        | Valsartan & Carbamazepine       |
| SRR16771870  | Gill        | Synthetic 17 a-Ethinylestradiol |
| SRR7725722   | Gill        | Diarrhetic Shellfish Poisoning  |
| SRR13013756  | Gill        | Hypoxia                         |


## Methods
## {auto-animate=true}
**Methods**

Choose organism & question

## {auto-animate=true}
**Methods**

Search NCBI for existing RNASeq files

Choose organism & question

## {auto-animate=true}
**Methods**

Align sequences using Kallisto

Search NCBI for existing RNASeq files

Choose organism & question

## {auto-animate=true}
**Methods**

Quantify DEG & Annotate

Align sequences using Kallisto

Search NCBI for existing RNASeq files

Choose organism & question

## {auto-animate=true}
**Methods**

Visualize Data

Quantify DEG & Annotate 

Align sequences using Kallisto

Search NCBI for existing RNASeq files

Choose organism & question

## {auto-animate=true}
**Methods**

Perform Gene Enrichment Analysis

Visualize Data

Quantify DEG & Annotate

Align sequences using Kallisto

Search NCBI for existing RNASeq files

Choose organism & question

## Preliminary Results {.smaller}

# Taking a look at my Kallisto output {.smaller}
countmatrix <- read.delim("../output/kallisto_01.isoform.counts.matrix", header = TRUE, sep = '\t')
rownames(countmatrix) <- countmatrix$X
countmatrix <- countmatrix[,-1]
head(countmatrix)

```{r}

# Taking a look at my Kallisto output
countmatrix <- read.delim("../output/kallisto_01.isoform.counts.matrix", header = TRUE, sep = '\t')
rownames(countmatrix) <- countmatrix$X
countmatrix <- countmatrix[,-1]
head(countmatrix)
#![kallisto](kallistoTable.png)
#trying to add the image but it isn't rendering

```

# Only 2 columns? {.smaller}
countmatrix <- read.delim("../output/kallisto_01.isoform.counts.matrix", header = TRUE, sep = '\t')
rownames(countmatrix) <- countmatrix$X
countmatrix <- countmatrix[,-1]
head(countmatrix)
dim(countmatrix)

```{r}
# Only 2 columns of values?
countmatrix <- read.delim("../output/kallisto_01.isoform.counts.matrix", header = TRUE, sep = '\t')
rownames(countmatrix) <- countmatrix$X
countmatrix <- countmatrix[,-1]
head(countmatrix)
dim(countmatrix)
```

## Troubleshooting Missing Sequences {.smaller}
```{.python code-line-numbers="1, 8"}
This code finds files ending in 1.fstq

find /home/shared/8TB_HDD_02/cnmntgna/GitHub/chris-musselcon/output/ncbi/*_1.fastq \
| xargs basename -s _1.fastq  | xargs -I{} /home/shared/kallisto/kallisto \
quant -i ../data/MGAL_cds.index \
-o ../output/kallisto_01/{} \
-t 4 \
/home/shared/8TB_HDD_02/cnmntgna/GitHub/chris-musselcon/output/ncbi/{}_1.fastq \
/home/shared/8TB_HDD_02/cnmntgna/GitHub/chris-musselcon/output/ncbi/{}_2.fastq \
```


```{.python code-line-numbers="1, 9"}
This code finds files ending in 2.fstq

find /home/shared/8TB_HDD_02/cnmntgna/GitHub/chris-musselcon/output/ncbi/*_1.fastq \
| xargs basename -s _1.fastq  | xargs -I{} /home/shared/kallisto/kallisto \
quant -i ../data/MGAL_cds.index \
-o ../output/kallisto_01/{} \
-t 4 \
/home/shared/8TB_HDD_02/cnmntgna/GitHub/chris-musselcon/output/ncbi/{}_1.fastq \
/home/shared/8TB_HDD_02/cnmntgna/GitHub/chris-musselcon/output/ncbi/{}_2.fastq \
```

## Next Steps
![mussels](https://github.com/course-fish546-2023/chris-musselcon/blob/main/assets/mytilusReef.png?raw=true)  
- Run 2 missing sequences through Kallisto  
- Run DESeq  
- Quantify DEG & Annotate  
- Visualize Data  
- Perform Gene Enrichment Analysis  

