---
title: "RNA-seq Data Retrieval & Kallisto Evaluation"
author: Chris
date: "`r format(Sys.time(), '%d %B, %Y')`"  
output: 
  html_document:
    theme: readable
    highlight: zenburn
    toc: true
    toc_float: true
    number_sections: true
    code_folding: show
    code_download: true
---

```{r setup, include=FALSE}
library(knitr)
library(tidyverse)
library(kableExtra)
library(DESeq2)
library(pheatmap)
library(RColorBrewer)
library(data.table)
library(DT)
library(Biostrings)
knitr::opts_chunk$set(
  echo = TRUE,         # Display code chunks
  eval = FALSE,         # Evaluate code chunks
  warning = FALSE,     # Hide warnings
  message = FALSE,     # Hide messages
  fig.width = 6,       # Set plot width in inches
  fig.height = 4,      # Set plot height in inches
  fig.align = "center" # Align plots to the center
)
```


**Data**\
Existing RNA-Seq data was retrieved from the following complete RNASeq data available in the [NCBI database](https://www.ncbi.nlm.nih.gov/):\
- SRR19782039 [Exposure to Valsartan & Carbamazepine](https://www.ncbi.nlm.nih.gov/sra/SRX15826291%5Baccn%5D)\
- SRR16771870 [Exposure to a synthetic hormone 17 a-Ethinylestradiol (EE2)](https://www.ncbi.nlm.nih.gov/sra/SRX12971792%5Baccn%5D)\
- SRR7725722 [Diarrhetic Shellfish Poisoning (DSP) toxins associated with Harmful Algal Blooms (HABs)](https://www.ncbi.nlm.nih.gov/sra/SRX4582204%5Baccn%5D)\
- SRR13013756 [Hypoxia](https://www.ncbi.nlm.nih.gov/sra/SRX9464766%5Baccn%5D)\
- *Mytilus galloprovincialis* [Reference Genome](https://www.ncbi.nlm.nih.gov/data-hub/genome/GCA_900618805.1/)

```{r}
getwd()
```

### Pulling my reference genome from prefab code provided by NCBI
curled genome and copied just cds file

```{r, engine='bash', eval=TRUE}

 cd /home/shared/8TB_HDD_02/cnmntgna/GitHub/chris-musselcon/data/ncbi_dataset/data/GCA_900618805.1
 head cds_from_genomic.fna
 
```



# Create the index file to align my short read files to the genes from the MGAL_10
```{r, engine='bash'}
/home/shared/kallisto/kallisto \
index \
-i ../data/MGAL_cds.index \
../data//ncbi_dataset/data/GCA_900618805.1/cds_from_genomic.fna
```

# Where is data


#Runnning Kallisto

```{r, engine='bash', eval=TRUE}
/home/shared/kallisto/kallisto quant
```

```{r, engine='bash', eval=TRUE}
ls /home/shared/8TB_HDD_02/cnmntgna/GitHub/chris-musselcon/output/ncbi/
```


```{r, engine='bash'}

#mkdir ../output
mkdir ../output/kallisto_01

#OLD CODE

find /home/shared/8TB_HDD_02/cnmntgna/GitHub/chris-musselcon/output/ncbi/*_1.fastq \
| xargs basename -s _1.fastq  | xargs -I{} /home/shared/kallisto/kallisto \
quant -i ../data/MGAL_cds.index \
-o ../output/kallisto_01/{} \
-t 4 \
/home/shared/8TB_HDD_02/cnmntgna/GitHub/chris-musselcon/output/ncbi/{}_1.fastq \
/home/shared/8TB_HDD_02/cnmntgna/GitHub/chris-musselcon/output/ncbi/{}_2.fastq \


```

Definitely missing the point of this structure
```{r, eval=FALSE, engine='bash'}
#Running for .fastq to pick up the other two, since i missed that not all had a 1 or 2
#how do i get *.fastq files as well? I thought I understood the syntax, but I am still getting it wrong
find /home/shared/8TB_HDD_02/cnmntgna/GitHub/chris-musselcon/output/ncbi/.fastq \
| xargs basename -s .fastq  | xargs -I{} /home/shared/kallisto/kallisto \
quant -i ../data/MGAL_cds.index \
-o ../output/kallisto_01/{} \
-t 4 \
/home/shared/8TB_HDD_02/cnmntgna/GitHub/chris-musselcon/output/ncbi/{}.fastq
```
#DESeq from Kallisto output
#trouble calling the program from the whole chunk below so I separated it to make sure I'm going to the right place.
```{r}
getwd()

```
```{r, engine='bash'}
cd /home/shared/trinityrnaseq-v2.12.0
```

#not able to run this without the full path - pretty sure i am missing something there, but this is the solution now.
```{r, engine='bash'}

perl /home/shared/trinityrnaseq-v2.12.0/util/abundance_estimates_to_matrix.pl \
--est_method kallisto \
    --gene_trans_map none \
    --out_prefix /home/shared/8TB_HDD_02/cnmntgna/GitHub/chris-musselcon/output/kallisto_01 \
    --name_sample_by_basedir \
    /home/shared/8TB_HDD_02/cnmntgna/GitHub/chris-musselcon/output/kallisto_01/SRR16771870/abundance.tsv \
    /home/shared/8TB_HDD_02/cnmntgna/GitHub/chris-musselcon/output/kallisto_01/SRR7725722/abundance.tsv \

```

```{r}

countmatrix <- read.delim("../output/kallisto_01.isoform.counts.matrix", header = TRUE, sep = '\t')
rownames(countmatrix) <- countmatrix$X
countmatrix <- countmatrix[,-1]
head(countmatrix)
tail(countmatrix)
dim(countmatrix)

```

```{r}
countmatrix <- round(countmatrix, 0)
str(countmatrix)

```

```{r}

dim(countmatrix)

dim(deseq2.colData)

```

```{r}

length(colnames(data))

```

```{r}
deseq2.colData <- data.frame(condition=factor(c("EE2", "DSP"))), 
                             type=factor(rep("single-read", 24)))
rownames(deseq2.colData) <- colnames(data)
deseq2.dds <- DESeqDataSetFromMatrix(countData = countmatrix,
                                     colData = deseq2.colData, 
                                     design = ~ condition)


























